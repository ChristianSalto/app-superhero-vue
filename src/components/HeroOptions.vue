<template>
  <div class="options-container">
    <ul>
      <li
        v-for="hero in heros"
        :key="hero.id"
        @click="$emit('onSelect', hero.id)"
      >
        {{ hero.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    heros: {
      type: Array,
      requiered: true,
    },
  },
};
</script>
<style scoped>
ul {
  list-style-type: none;
  padding: inherit;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
  padding: 10px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>
