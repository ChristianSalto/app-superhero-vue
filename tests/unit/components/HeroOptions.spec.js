import { shallowMount } from '@vue/test-utils'
import HeroOptions from '@/components/HeroOptions'

import { heros } from '../mocks/heros.mock'


describe('HeroOptions Component', () => {

    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(HeroOptions, {
            props: {
                heros
            }
        })
    })

    test('debe hacer match con el snapshot', () => {
        // console.log(wrapper.html())
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe de mostrar las 4 opciones correctamente', () => {
        const liArray = wrapper.findAll('li');
        expect(liArray.length).toBe(4)

        expect(liArray[0].text()).toBe('A-Bomb')
        expect(liArray[1].text()).toBe('Abe Sapien')
        expect(liArray[2].text()).toBe('Abin Sur')
        expect(liArray[3].text()).toBe('Abomination')

    })

    test('debe de emitir "selection" con sus respectivos parametros al hacer click', () => {
        const [li1, li2, li3, li4] = wrapper.findAll('li');

        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')

        console.log(wrapper.emitted('onSelect'));

        expect( wrapper.emitted('onSelect').length ).toBe(4)
        expect( wrapper.emitted('onSelect')[0] ).toEqual(["1"])
        expect( wrapper.emitted('onSelect')[1] ).toEqual(["2"])
        expect( wrapper.emitted('onSelect')[2] ).toEqual(["3"])
        expect( wrapper.emitted('onSelect')[3] ).toEqual(["4"])
    })
})