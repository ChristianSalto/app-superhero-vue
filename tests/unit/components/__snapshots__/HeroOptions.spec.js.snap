// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroOptions Component debe hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>A-Bomb</li>
    <li>Abe Sapien</li>
    <li>Abin Sur</li>
    <li>Abomination</li>
  </ul>
</div>
`;
